### PR TITLE
add hamburger nav menu

### PR DESCRIPTION
**Pull request by 5h1rU**
TODO: 
- [x] remove old links
- [x] add hamburger menu with custom caret functionality
- [x] align search bar and navigation buttons
- [x] modify the LinkWithIconOnlyTooltip functionality
- [x] modify the LinkWithIconOnlyTooltip name
- [x] customize the dropdown min-width
- [x] add MenuNavItem component
- [x] add test for MenuNavItem component
- [x] adjust user menu size
- [x] fix StatusMessagesNavItem styles
- [x] fix broken tests
- [x] Design is good

### CODE REVIEW

**Review by tjkandala — COMMENTED** [review]
Code looks good, but shouldn't the menu items collapse into the hamburger menu at 1025px width? It seems like we always show the hamburger menu in this implementation

File: client/web/src/nav/MenuNavItem.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import React, { useCallback, useState } from 'react'
+import { ButtonDropdown, DropdownItem, DropdownMenu, DropdownToggle } from 'reactstrap'
+import MenuDownIcon from 'mdi-react/MenuDownIcon'
+import MenuUpIcon from 'mdi-react/MenuUpIcon'
+
+interface MenuNavItemProps {
+    children: React.ReactNode
+}
+
+export const MenuNavItem: React.FunctionComponent<MenuNavItemProps> = props => {
```
Comment (felixfbecker) [inline client/web/src/nav/MenuNavItem.tsx:11]: **suggestion**: maybe add a docblock explaining what this is, i.e. a dropdown menu for use in the navbar?

**Review by felixfbecker — COMMENTED** [review]
LGTM otherwise!
Also left comments in Chromatic

**Comment by 5h1rU** [comment]
> Code looks good, but shouldn't the menu items collapse into the hamburger menu at 1025px width? It seems like we always show the hamburger menu in this implementation (note: I may be missing context, I'm just looking at Figma and the linked GitHub issue)

Thanks for pointing this out @tjkandala 
This was already discussed with @AlicjaSuska Now, we are using `< 992px` as the width to show the hamburger menu

File: client/web/src/nav/NavLinks.tsx [hunk 2]
```diff
@@ -71,22 +72,35 @@ export class NavLinks extends React.PureComponent<Props> {
...
+                    <MenuNavItem>
+                        {!isErrorLike(this.props.settingsCascade.final) &&
+                            this.props.settingsCascade.final?.experimentalFeatures?.codeInsights &&
+                            !this.props.minimalNavLinks && <InsightsNavItem />}
+
+                        {!isErrorLike(this.props.settingsCascade.final) &&
+                            this.props.settingsCascade.final?.experimentalFeatures?.codeMonitoring &&
+                            !this.props.minimalNavLinks && <CodeMonitoringNavItem />}
+
+                        {!this.props.minimalNavLinks && this.props.showCampaigns && <CampaignsNavItem />}
```
[the remark this answers is not in the document]
Comment (felixfbecker) [reply client/web/src/nav/NavLinks.tsx]: Sorry for being unclear - what I meant to suggest was not to extract these into a method (that still evaluates the condition multiple times) but to use nested conditions like this:
```ts
{!this.props.minimalNavLinks && <>
  	{!isErrorLike(this.props.settingsCascade.final) && <>
	  		{this.props.settingsCascade.final?.experimentalFeatures?.codeInsights && <InsightsNavItem />}
	  		{this.props.settingsCascade.final?.experimentalFeatures?.codeMonitoring && <CodeMonitoringNavItem />}
	</>}
	{this.props.showCampaigns && <CampaignsNavItem />}
</>}
```

Note how each condition is only checked exactly once and it is easier to see at a glance which items are disabled/enabled by e.g. the `minimalNavLinks` prop.

**Review by AlicjaSuska — CHANGES_REQUESTED** [review]
Great work @5h1rU ! The only thing missing is the chevron position change. When menu is closed, it should point down. When it's open, it should point up.

Screenshot 2021-01-27 at 16 49 10

**Comment by 5h1rU** [comment]
@AlicjaSuska dropdown behavior updated :)

File: client/web/src/nav/NavLinks.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (5h1rU) [reply client/web/src/nav/NavLinks.tsx]: Got it!

**Comment by felixfbecker** [comment]
@AlicjaSuska that change looked unintentional/inconsistent with the dropdowns in the component library in Figma to me. I had pointed it out to Felipe here: 
https://www.chromatic.com/pullrequest?activeElementId=comparison-test-key-5f0f3b9bb320641bb08ced08-1200&appId=5f0f381c0e50750022dc6bf7&number=17535&view=changes

**Comment by felixfbecker** [comment]
@AlicjaSuska Should the dropdown caret positions be updated for all dropdowns (the generic dropdown component)? It seems inconsistent to have them different only in the navbar, and the implementation (the rendering of the different carets) is now duplicated in every dropdown instead of just using `caret={true}`. If we want it to switch I think we should abstract it in a component, but that seems a bit out of scope for this PR.

**Comment by AlicjaSuska** [comment]
I think carret should change its direction in all cases - it's better ux, carret state indicates that something changed and in which direction the movement will happen on click. @5h1rU can we change it for the nav bar items for now so they are all consistent with the current implementation of the command palette? We can take care of all other dropwdons in a separate issue @felixfbecker wdyt?

**Review by felixfbecker — CHANGES_REQUESTED** [review]
Let's update the other dropdowns in a separate PR (should be abstracted somehow, e.g. in a component or with CSS like `.dropdown-toggle[aria-expanded="true"]::after { rotate: 180deg; }`).

Other than that looks great!

**Review by felixfbecker — APPROVED** [review]
Woops that was meant to be an approval

**Comment by AlicjaSuska** [comment]
@5h1rU I found one more visual correction to make:

- when hovering on the items in the menu, only one item row background should be highlighted. Instead, all rows change the background. Text color changes correctly on hover. It happens for both light and dark themes.

Screenshot 2021-01-29 at 15 18 24

Should be the same as in the usernav menu:
Screenshot 2021-01-29 at 15 20 52

**Review by AlicjaSuska — CHANGES_REQUESTED** [review]
Row highlight on hover described in the comment above.

**Comment by 5h1rU** [comment]
> Let's update the other dropdowns in a separate PR (should be abstracted somehow, e.g. in a component or with CSS like `.dropdown-toggle[aria-expanded="true"]::after { rotate: 180deg; }`).
> 
> Other than that looks great!

Thanks, @felixfbecker. I'd love to create a new component for this purpose, the CSS solution is a good one, but I don't feel comfortable manipulating the Reactstrap core so much. The less custom CSS, the better.

**Comment by 5h1rU** [comment]
@AlicjaSuska The problem with the hover status was addressed. Thanks for letting me know.

**Review by AlicjaSuska — APPROVED** [review]
LGTM! thank you @5h1rU

**Comment by AlicjaSuska** [comment]
@5h1rU could you update the Figma Design link in Storybook to (this will link to the design system file, instead of the project file). Thank you!

https://www.figma.com/file/SFhXbl23TJ2j5tOF51NDtF/%F0%9F%93%9AWeb?node-id=1108%3A872

**Pull request merged**